src/main/java/ee/tuleva/onboarding/epis/mandate/details/MandateDetailsDeserializer.java

```diff
@@ -19,16 +19,23 @@ public Object findDeserializer(Annotated a) {
         }
       };
 
+  public static MandateType deserializeMandateTypeField(JsonNode mandateDetailsRoot)
+      throws IOException {
+    MandateType type = MandateType.valueOf(mandateDetailsRoot.get("mandateType").asText());
+
+    if (type == MandateType.UNKNOWN) {
+      throw new IllegalArgumentException("Unknown mandateType: " + type);
+    }
+
+    return type;
+  }
+
   @Override
   public MandateDetails deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
     ObjectMapper mapper = (ObjectMapper) p.getCodec();
     JsonNode root = mapper.readTree(p);
 
-    MandateType type = MandateType.valueOf(root.get("mandateType").asText());
-
-    if (type == MandateType.UNKNOWN) {
-      throw new IllegalArgumentException("Unknown mandateType: " + type);
-    }
+    MandateType type = MandateDetailsDeserializer.deserializeMandateTypeField(root);
```

Comment:
You can do just `deserializeMandateTypeField(root)` since it's in the same class.